<?xml version="1.0" ?>
<!-- ========================================================================= -->
<!-- |    This document was autogenerated by xacro from kinova.urdf.xacro    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                       | -->
<!-- ========================================================================= -->
<robot name="Gen3">
	<!-- create link fixed to the "world" -->
	<link name="world"/>
	<joint name="base_joint" type="fixed">
		<origin rpy="0 0 0" xyz="0 0 0"/>
		<parent link="world"/>
		<child link="base_link"/>
	</joint>
	<link name="base_link">
		<inertial>
			<origin rpy="0 0 0" xyz="-0.000648 -0.000166 0.084487"/>
			<mass value="1.697"/>
			<inertia ixx="0.004622" ixy="9E-06" ixz="6E-05" iyy="0.004495" iyz="9E-06" izz="0.002079"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/base_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/base_link.STL"/>
			</geometry>
		</collision>
	</link>
	<link name="shoulder_link">
		<inertial>
			<origin rpy="0 0 0" xyz="-2.3E-05 -0.010364 -0.07336"/>
			<mass value="1.377"/>
			<inertia ixx="0.00457" ixy="1E-06" ixz="2E-06" iyy="0.004831" iyz="0.000448" izz="0.001409"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/shoulder_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/shoulder_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_1" type="continuous">
		<origin rpy="-3.1416 0.0 0.0" xyz="0 0 0.15643"/>
		<parent link="base_link"/>
		<child link="shoulder_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="39" velocity="1.3963"/>
	</joint>
	<link name="bicep_link">
		<inertial>
			<origin rpy="0 0 0" xyz="3.5E-05 -0.208207 -0.01889"/>
			<mass value="1.262"/>
			<inertia ixx="0.046752" ixy="-9E-06" ixz="0" iyy="0.00085" iyz="-9.8E-05" izz="0.047188"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/bicep_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/bicep_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_2" type="revolute">
		<origin rpy="1.5708 0.0 0.0" xyz="0 0.005375 -0.12838"/>
		<parent link="shoulder_link"/>
		<child link="bicep_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="39" lower="-2.41" upper="2.41" velocity="1.3963"/>
	</joint>
	<link name="forearm_link">
		<inertial>
			<origin rpy="0 0 0" xyz="1.8E-05 0.076168 -0.01397"/>
			<mass value="0.93"/>
			<inertia ixx="0.008292" ixy="-1E-06" ixz="0" iyy="0.000628" iyz="0.000432" izz="0.008464"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/forearm_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/forearm_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_3" type="revolute">
		<origin rpy="3.1416 0 0" xyz="0 -0.41 0"/>
		<parent link="bicep_link"/>
		<child link="forearm_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="39" lower="-2.66" upper="2.66" velocity="1.3963"/>
	</joint>
	<link name="spherical_wrist_1_link">
		<inertial>
			<origin rpy="0 0 0" xyz="-1E-06 0.008486 -0.062937"/>
			<mass value="0.6781"/>
			<inertia ixx="0.001645" ixy="0" ixz="0" iyy="0.001666" iyz="-0.000234" izz="0.000389"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/spherical_wrist_1_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/spherical_wrist_1_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_4" type="continuous">
		<origin rpy="1.5708 0.0 0.0" xyz="0 0.20843 -0.006375"/>
		<parent link="forearm_link"/>
		<child link="spherical_wrist_1_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="9" velocity="1.2218"/>
	</joint>
	<link name="spherical_wrist_2_link">
		<inertial>
			<origin rpy="0 0 0" xyz="-1E-06 0.046429 -0.008704"/>
			<mass value="0.678"/>
			<inertia ixx="0.001685" ixy="0" ixz="0" iyy="0.0004" iyz="0.000255" izz="0.001696"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/spherical_wrist_2_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/spherical_wrist_2_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_5" type="revolute">
		<origin rpy="-1.5708 0.0 0.0" xyz="0 -0.00017505 -0.10593"/>
		<parent link="spherical_wrist_1_link"/>
		<child link="spherical_wrist_2_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="9" lower="-2.23" upper="2.23" velocity="1.2218"/>
	</joint>
	<!-- <xacro:if value="${vision}"> -->
	<link name="bracelet_link">
		<inertial>
			<origin rpy="0 0 0" xyz="0.000281 0.011402 -0.029798"/>
			<mass value="0.5"/>
			<inertia ixx="0.000587" ixy="3E-06" ixz="3E-06" iyy="0.000369" iyz="-0.000118" izz="0.000609"/>
		</inertial>
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/bracelet_with_vision_link.STL"/>
			</geometry>
			<material name="">
				<color rgba="0.75294 0.75294 0.75294 1"/>
			</material>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://kortex_description/arms/gen3/6dof/meshes/bracelet_with_vision_link.STL"/>
			</geometry>
		</collision>
	</link>
	<joint name="joint_6" type="continuous">
		<origin rpy="1.5708 0.0 0.0" xyz="0 0.10593 -0.00017505"/>
		<parent link="spherical_wrist_2_link"/>
		<child link="bracelet_link"/>
		<axis xyz="0 0 1"/>
		<limit effort="9" velocity="1.2218"/>
	</joint>
	<link name="end_effector_link"/>
	<joint name="end_effector" type="fixed">
		<origin rpy="3.14159265358979 1.09937075168372E-32 3.14159265358979" xyz="0 0 -0.0615250000000001"/>
		<parent link="bracelet_link"/>
		<child link="end_effector_link"/>
		<axis xyz="0 0 0"/>
	</joint>
	<!-- TODO(destogl) this probably collides with camera URDF -->
	<!-- <xacro:if value="${vision}"> -->
	<link name="camera_link"/>
	<joint name="camera_module" type="fixed">
		<origin rpy="3.14159265358979 3.14159265358979 0" xyz="0 0.05639 -0.00305"/>
		<parent link="end_effector_link"/>
		<child link="camera_link"/>
	</joint>
	<link name="camera_depth_frame"/>
	<joint name="depth_module" type="fixed">
		<origin rpy="3.14159265358979 3.14159265358979 0" xyz="0.0275 0.066 -0.00305"/>
		<parent link="end_effector_link"/>
		<child link="camera_depth_frame"/>
	</joint>
	<link name="camera_color_frame"/>
	<joint name="color_module" type="fixed">
		<origin rpy="3.14159265358979 3.14159265358979 0" xyz="0 0.05639 -0.00305"/>
		<parent link="end_effector_link"/>
		<child link="camera_color_frame"/>
	</joint>
	<!-- </xacro:if> -->
	<!-- this is a temporary link to rotate the 2f-140 gripper to match the 2f-85 -->
	<link name="robotiq_base_link"/>
	<joint name="robotiq_base_joint" type="fixed">
		<parent link="end_effector_link"/>
		<child link="robotiq_base_link"/>
		<origin rpy="0 0 0" xyz="0 0 0"/>
	</joint>
  <link name="robotiq_85_base_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/robotiq_base.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/robotiq_base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 2.274e-05 0.03232288"/>
      <mass value="6.6320197e-01"/>
      <inertia ixx="5.1617816e-04" ixy="2.936e-8" ixz="0.0" iyy="5.8802208e-04" iyz="-3.2296e-7" izz="3.9462776e-04"/>
    </inertial>
  </link>
	<joint name="robotiq_85_base_joint" type="fixed">
		<parent link="robotiq_base_link"/>
		<child link="robotiq_85_base_link"/>
		<origin rpy="0 0 0" xyz="0 0 0"/>
	</joint>
  <link name="robotiq_85_left_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/left_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/left_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01213197 0.0002 -0.00058647"/>
      <mass value="1.384773208e-02"/>
      <inertia ixx="3.5232e-7" ixy="0.0" ixz="1.1744e-7" iyy="2.31944e-6" iyz="0" izz="2.23136e-6"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/right_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/right_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01213197 0.0002 -0.00058647"/>
      <mass value="1.384773208e-02"/>
      <inertia ixx="3.5232e-7" ixy="0.0" ixz="-1.1744e-7" iyy="2.31944e-6" iyz="0.0" izz="2.23136e-6"/>
    </inertial>
  </link>
  <link name="robotiq_85_left_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/left_finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/left_finger.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.00346899 -0.00079447 0.01867121"/>
      <mass value="4.260376752e-02"/>
      <inertia ixx="1.385792000000000e-05" ixy="0.0" ixz="-2.17264e-06" iyy="1.183208e-05" iyz="0.0" izz="5.19672e-06"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/right_finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/right_finger.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00346899 -5.53e-06 0.01867121"/>
      <mass value="4.260376752000000e-02"/>
      <inertia ixx="1.385792e-05" ixy="0.0" ixz="2.17264e-06" iyy="1.183208e-05" iyz="0.0" izz="5.19672e-06"/>
    </inertial>
  </link>
  <link name="robotiq_85_left_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/left_inner_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/left_inner_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01897699 0.00015001 0.02247101"/>
      <mass value="2.969376448e-02"/>
      <inertia ixx="9.57136e-06" ixy="0.0" ixz="-3.93424e-06" iyy="8.69056e-06" iyz="0.0" izz="8.19144e-06"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_inner_knuckle_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/right_inner_knuckle.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/right_inner_knuckle.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01926824 5.001e-05 0.02222178"/>
      <mass value="2.969376448e-02"/>
      <inertia ixx="9.42456e-06" ixy="0.0" ixz="3.9636e-06" iyy="8.69056e-06" iyz="0.0" izz="8.33824e-06"/>
    </inertial>
  </link>
  <link name="robotiq_85_left_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/left_finger_tip.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/left_finger_tip.stl"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu1>100000.0</mu1>
            <mu2>100000.0</mu2>
          </ode>
        </friction>
        <contact>
          <ode>
            <kp>1e+5</kp>
            <kd>1</kd>
            <soft_cfm>0</soft_cfm>
            <soft_erp>0.2</soft_erp>
            <minDepth>0.002</minDepth>
            <maxVel>0</maxVel>
          </ode>
        </contact>
      </surface>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.01456706 -0.0008 0.01649701"/>
      <mass value="4.268588744e-02"/>
      <inertia ixx="1.048152e-05" ixy="0.0" ixz="3.5232e-6" iyy="1.197888e-05" iyz="0.0" izz="4.22784e-06"/>
    </inertial>
  </link>
  <link name="robotiq_85_right_finger_tip_link">
    <visual>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/visual/2f_85/right_finger_tip.dae"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robotiq_description/meshes/collision/2f_85/right_finger_tip.stl"/>
      </geometry>
      <surface>
        <friction>
          <ode>
            <mu1>100000.0</mu1>
            <mu2>100000.0</mu2>
          </ode>
        </friction>
        <contact>
          <ode>
            <kp>1e+5</kp>
            <kd>1</kd>
            <soft_cfm>0</soft_cfm>
            <soft_erp>0.2</soft_erp>
            <minDepth>0.002</minDepth>
            <maxVel>0</maxVel>
          </ode>
        </contact>
      </surface>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.01456706 5e-05 0.01649701"/>
      <mass value="4.268588744e-02"/>
      <inertia ixx="1.048152e-05" ixy="0.0" ixz="-3.5232e-06" iyy="1.197888e-05" iyz="0.0" izz="4.22784e-06"/>
    </inertial>
  </link>
  <!-- <joint name="robotiq_85_left_knuckle_joint" type="revolute"> -->
  <joint name="left_finger_joint" type="revolute">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_knuckle_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="0.03060114 0.0 0.05490452"/>
    <limit effort="50" lower="0.0" upper="0.8" velocity="0.5"/>
  </joint>
  <joint name="robotiq_85_right_knuckle_joint" type="revolute">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_knuckle_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="-0.03060114 0.0 0.05490452"/>
    <limit effort="50" lower="-0.8" upper="0.0" velocity="0.5"/>
    <mimic joint="left_finger_joint" multiplier="-1"/>
  </joint>
  <joint name="robotiq_85_left_finger_joint" type="fixed">
    <parent link="robotiq_85_left_knuckle_link"/>
    <child link="robotiq_85_left_finger_link"/>
    <origin rpy="0 0 0" xyz="0.03152616 0.0 -0.00376347"/>
  </joint>
  <joint name="robotiq_85_right_finger_joint" type="fixed">
    <parent link="robotiq_85_right_knuckle_link"/>
    <child link="robotiq_85_right_finger_link"/>
    <origin rpy="0 0 0" xyz="-0.03152616 0.0 -0.00376347"/>
  </joint>
  <joint name="robotiq_85_left_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_left_inner_knuckle_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="0.0127 0.0 0.06142"/>
    <mimic joint="left_finger_joint"/>
  </joint>
  <joint name="robotiq_85_right_inner_knuckle_joint" type="continuous">
    <parent link="robotiq_85_base_link"/>
    <child link="robotiq_85_right_inner_knuckle_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="-0.0127 0.0 0.06142"/>
    <mimic joint="left_finger_joint" multiplier="-1"/>
  </joint>
  <joint name="robotiq_85_left_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_left_finger_link"/>
    <child link="robotiq_85_left_finger_tip_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="0.00563134 0.0 0.04718515"/>
    <mimic joint="left_finger_joint" multiplier="-1"/>
  </joint>
  <joint name="robotiq_85_right_finger_tip_joint" type="continuous">
    <parent link="robotiq_85_right_finger_link"/>
    <child link="robotiq_85_right_finger_tip_link"/>
    <axis xyz="0 -1 0"/>
    <origin rpy="0 0 0" xyz="-0.00563134 0.0 0.04718515"/>
    <mimic joint="left_finger_joint"/>
  </joint>
	<link name="dummy_right_finger_link"/>
	<joint name="right_finger_joint" type="fixed">
		<parent link="robotiq_85_base_link"/>
		<child link="dummy_right_finger_link"/>
	</joint>
</robot>
